src/fsearch_preview.c

```diff
@@ -0,0 +1,88 @@
+/*
+   FSearch - A fast file search utility
+   Copyright © 2020 Christian Boxdörfer
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include "fsearch_preview.h"
+#include "fsearch.h"
+
+#include <gio/gio.h>
+
+#define PREVIEWER_DBUS_NAME "org.gnome.NautilusPreviewer"
+#define PREVIEWER_DBUS_IFACE "org.gnome.NautilusPreviewer"
+#define PREVIEWER_DBUS_PATH "/org/gnome/NautilusPreviewer"
+
+static void
+preview_show_file_ready_cb(GObject *source, GAsyncResult *res, gpointer user_data) {
+    GError *error = NULL;
```

Comment:
We can use the `g_autoptr` macro here for `GError` to avoid the manual `g_error_free`